File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Tanimlamalar/Diagnoses/DiagnosesColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace MuayeneYonetimPortali.Tanimlamalar.Columns;

[ColumnsScript("Tanimlamalar.Diagnoses")]
[BasedOnRow(typeof(DiagnosesRow), CheckNames = true)]
public class DiagnosesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int DiagnosisId { get; set; }
    [EditLink]
    public string Code { get; set; }
    public string Name { get; set; }
    public bool IsActive { get; set; }
}

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Tanimlamalar/DiagnosesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DiagnosesRow } from "./DiagnosesRow";

export interface DiagnosesColumns {
    DiagnosisId: Column<DiagnosesRow>;
    Code: Column<DiagnosesRow>;
    Name: Column<DiagnosesRow>;
    IsActive: Column<DiagnosesRow>;
}

export class DiagnosesColumns extends ColumnsBase<DiagnosesRow> {
    static readonly columnsKey = 'Tanimlamalar.Diagnoses';
    static readonly Fields = fieldsProxy<DiagnosesColumns>();
}